public/app/features/variables/query/actions.ts

```diff
@@ -27,10 +27,7 @@ export const updateQueryVariableOptions = (
         dispatch(removeVariableEditorError({ errorProp: 'update' }));
       }
       const dataSource = await getDatasourceSrv().get(variableInState.datasource ?? '');
-      const queryOptions: any = { range: undefined, variable: variableInState, searchFilter };
-      if (variableInState.refresh === VariableRefresh.onTimeRangeChanged) {
-        queryOptions.range = getTimeSrv().timeRange();
-      }
+      const queryOptions: any = { range: getTimeSrv().timeRange(), variable: variableInState, searchFilter };
```

Comment:
we need to add refresh: variableInState.refresh   here as well so data source knows if it should use current time range in variable name looks or not, this change could have some big breaking behavior so I am not 100% this is correct do to yet

Reply:
Hey, thanks for having a look at this :- )
Excuse a maybe oblivious question, but my understanding would be that variableInState.refresh is returned regardless of if this conditional is present?
But it seems the tests are failing and it is looking like this change is a bit more involved, as you say. I'm in no rush to push this through, but I was hoping to at least get the issue on your radar. I don't have the resources right now, to do a more involved fix, but I may look at it at a later date, unless someone else beats me to it.